### PR TITLE
feat(DIA-181): add tracking when tapping verified representative

### PR DESCRIPTION
**Pull request by araujobarret**
This PR resolves [DIA-181] 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add artist representative tracking - araujobarret

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-181]: https://artsyproduct.atlassian.net/browse/DIA-181?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/Artist/ArtistHeader.tsx [hunk 1]
```diff
@@ -80,9 +83,20 @@ export const ArtistHeader: React.FC<Props> = ({ artist, me, onLayoutChange }) =>
...
+    if (partner?.href && partner?.internalID) {
+      tracking.trackEvent({
+        action: ActionType.tappedVerifiedRepresentative,
+        context_module: ContextModule.artistHeader,
+        context_screen_owner_type: OwnerType.artist,
+        context_screen_owner_id: artist.internalID,
+        context_screen_owner_slug: artist.slug,
+        destination_page_owner_id: partner.internalID,
+        destination_page_owner_type: OwnerType.partner,
+      })
```
Comment (MounirDhahri) [inline src/app/Components/Artist/ArtistHeader.tsx:98]: **issue(minor):** Although this is kind of fine, I would suggest following the docs on best tracking practices in Eigen and moving the tracked event to a separate constant that we can then make sure is following a cohesion action type

**Review by MounirDhahri — CHANGES_REQUESTED** [review]

File: src/app/Components/Artist/ArtistHeader.tsx (hunk 1 above)
Comment (araujobarret) [reply src/app/Components/Artist/ArtistHeader.tsx:98]: oh I see, I'd expect the param type to be auto-checked by `trackEvent` typings, but I can do that

Comment (MounirDhahri) [reply src/app/Components/Artist/ArtistHeader.tsx:98]: Thank you 🙏

Comment (araujobarret) [reply src/app/Components/Artist/ArtistHeader.tsx]: was a very good request, done in 353444c8cc3897c8b11635b2e9dd6761efd46ec9

**Review by MounirDhahri — APPROVED** [review]
LGTM - thank you 🙏

File: src/app/Components/Artist/ArtistHeader.tests.tsx [hunk 2]
```diff
@@ -5,8 +5,10 @@ import { __globalStoreTestUtils__ } from "app/store/GlobalStore"
 import { navigate } from "app/system/navigation/navigate"
 import { setupTestWrapper } from "app/utils/tests/setupTestWrapper"
 import { graphql } from "react-relay"
+import { useTracking } from "react-tracking"
 
 describe("ArtistHeader", () => {
+  const tracking = useTracking().trackEvent
```
Comment (MounirDhahri) [inline src/app/Components/Artist/ArtistHeader.tests.tsx:11]: **suggestion:** I would suggest using  `mockTrackEvent` here for easier testing

Comment (araujobarret) [reply src/app/Components/Artist/ArtistHeader.tests.tsx]: done in [a88a5a4](https://github.com/artsy/eigen/pull/9373/commits/a88a5a482bcb10b274b80b227f37dde60ac14564)

**Pull request merged**